nexus/db-queries/tests/output/virtual_provisioning_collection_update_delete_instance.sql

```diff
@@ -40,25 +40,17 @@ WITH
             1
         )
         = 1
+        AND EXISTS(
+            SELECT 1 FROM instance WHERE instance.id = $5 AND instance.state_generation < $6 LIMIT 1
+          )
```

Comment:
The problem here is that (before this PR) `do_update` could be "true", but the state generation could be below our "max", which would result in that buggy partial deletion state.

Now:
- If `do_update` is true, we really believe we should perform the insertion/deletion
- If that's the case, we can use it as the deciding factor to both delete the `virtual_provisioning_resource` **and** decrement the usage in the `virtual_provisioning_collection` table